connectors/spark/src/test/java/io/unitycatalog/spark/ExternalTableReadWriteTest.java

```diff
@@ -19,34 +19,49 @@
 import io.unitycatalog.spark.utils.OptionsUtil;
 import java.io.File;
 import java.io.IOException;
-import java.time.Instant;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
-import org.apache.spark.network.util.JavaUtils;
-import org.apache.spark.sql.Dataset;
+import java.util.stream.Stream;
 import org.apache.spark.sql.Row;
 import org.apache.spark.sql.SparkSession;
 import org.apache.spark.sql.types.DataTypes;
 import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.CsvSource;
-
-public class TableReadWriteTest extends BaseSparkIntegrationTest {
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
+public class ExternalTableReadWriteTest extends BaseTableReadWriteTest {
   private static final String ANOTHER_PARQUET_TABLE = "test_parquet_another";
   private static final String PARQUET_TABLE_PARTITIONED = "test_parquet_partitioned";
   private static final String DELTA_TABLE = "test_delta";
   private static final String PARQUET_TABLE = "test_parquet";
   private static final String ANOTHER_DELTA_TABLE = "test_delta_another";
-  private static final String DELTA_TABLE_PARTITIONED = "test_delta_partitioned";
 
-  private final File dataDir = new File(System.getProperty("java.io.tmpdir"), "spark_test");
+  private File dataDir;
```

Comment:
Maybe you can use `@TempDir` annoation directly,  like this: https://github.com/unitycatalog/unitycatalog/blob/main/connectors/spark/src/test/java/io/unitycatalog/spark/auth/BaseCredRenewITTest.java#L66

Then you don't have to create an unique `dataDir` for every test method, since it's naturally unique generated after this annotation.